[AOTInductor] Propagate ConstantType for main graph.

muchulee8 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #152272

Summary:
We need to make sure all named_parameters and named_buffers be
propagated if we use runtime constant folding.

Test Plan:
python test/inductor/test_aot_inductor.py -k test_constant_type_propagation

Reviewers:

Subscribers:

Tasks:

Tags:

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @amjames @chauhang @aakhundov

muchulee8 committed:
[AOTInductor] Propagate ConstantType for main graph.

Summary:
We need to make sure all named_parameters and named_buffers be
propagated if we use runtime constant folding.

Test Plan:
python test/inductor/test_aot_inductor.py -k test_constant_type_propagation

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

Skylion007 commented on torch/_inductor/graph.py:
```diff
@@ -421,6 +427,10 @@ def __init__(
         # only keeping one node per device for stack trace purposes
         self.device_node_mapping: dict[torch.device, torch.fx.Node] = {}
         self.orig_gm: torch.fx.GraphModule = gm.__copy__()
+        for k, v in self.orig_gm.named_buffers():
```
Just use update to set all extend all the keys and values efficiently.

Skylion007 commented on torch/_inductor/graph.py:
```diff
@@ -421,6 +427,10 @@ def __init__(
         # only keeping one node per device for stack trace purposes
         self.device_node_mapping: dict[torch.device, torch.fx.Node] = {}
         self.orig_gm: torch.fx.GraphModule = gm.__copy__()
+        for k, v in self.orig_gm.named_buffers():
+            self.named_buffers[k] = v
```
```suggestion
        self.named_buffers.update(self.orig_gm.named_buffers())
```

Skylion007 commented on torch/_inductor/graph.py:
```diff
@@ -421,6 +427,10 @@ def __init__(
         self.orig_gm: torch.fx.GraphModule = gm.__copy__()
+        for k, v in self.orig_gm.named_buffers():
+            self.named_buffers[k] = v
+        for k, v in self.orig_gm.named_parameters():
+            self.named_parameters[k] = v
```
```suggestion
        self.named_parameters.update(self.orig_gm.named_parameters())
```

22quinn committed:
comment

Co-authored-by: Aaron Gokaslan <aaronGokaslan@gmail.com>

muchulee8 commented:
@pytorchbot merge

muchulee8 commented:
@muchulee8 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D73818587).

muchulee8 committed:
Update on "[AOTInductor] Propagate ConstantType for main graph."


Summary:
We need to make sure all named_parameters and named_buffers be
propagated if we use runtime constant folding.

Test Plan:
python test/inductor/test_aot_inductor.py -k test_constant_type_propagation

Reviewers:

Subscribers:

Tasks:

Tags:

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng amjames chauhang aakhundov

Differential Revision: [D73818587](https://our.internmc.facebook.com/intern/diff/D73818587)

[ghstack-poisoned]

muchulee8 commented:
@pytorchbot merge
